pkg/clustermesh/kvstoremesh/metrics.go

```diff
@@ -10,16 +10,18 @@ import (
 
 type Metrics struct {
 	// LeaderElectionStatus indicates state of leader election
-	LeaderElectionStatus metric.Vec[metric.Gauge]
+	LeaderElectionStatus metric.Gauge
 }
 
 func MetricsProvider() Metrics {
 	return Metrics{
+		// LeaderElectionStatus mimic similar metric in controller-manager
+		// by using the same metrics name and forcing a name label
 		LeaderElectionStatus: metric.NewGaugeVec(metric.GaugeOpts{
 			Namespace: metrics.Namespace,
 			Subsystem: "",
 			Name:      "leader_election_master_status",
-			Help:      "leader election status",
-		}, []string{metrics.LabelLeaderElectionName}),
+			Help:      "The leader election status",
+		}, []string{"name"}).WithLabelValues("kvstoremesh"),
```

Comment:
Nit. You could probably use `ConstLabels` to achieve the same result. The downside of both these approaches is that it would not be possible to reuse the metric for a different leader election in case we needed to introduce one (although I recon that it feels fairly unlikely).